parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetReader.java

```diff
@@ -114,6 +126,10 @@ private ParquetReader(Configuration conf,
     List<FileStatus> statuses = Arrays.asList(fs.listStatus(file, HiddenFileFilter.INSTANCE));
     List<Footer> footers = ParquetFileReader.readAllFootersInParallelUsingSummaryFiles(conf, statuses, false);
     this.footersIterator = footers.iterator();
+    globalMetaData = ParquetFileWriter.getGlobalMetaData(footers);
+    MessageType schema = globalMetaData.getSchema();
+    Map<String, Set<String>> extraMetadata = globalMetaData.getKeyValueMetaData();
+    readContext = readSupport.init(new InitContext(conf, extraMetadata, schema));
```

Comment:
this happens in reader.initialize in the record case. Let's make this consistent. It should happen in the same class in all cases.